### PR TITLE
Use custom cargo action for release

### PR DESCRIPTION
**Pull request by parfeon**
build(cargo): use custom cargo action for release

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -36,11 +36,22 @@ jobs:
...
-        run: ./.github/workflows/release/publish.sh
+          token: ${{ secrets.GH_TOKEN }}
+      - name: Checkout actions
+        uses: actions/checkout@v3
+        with:
+          repository: pubnub/client-engineering-deployment-tools
+          ref: v1
+          token: ${{ secrets.GH_TOKEN }}
+          path: .github/.release/actions
+      - name: Publish to CocoaPods Trunk
```
Comment (Xavrax) [inline .github/workflows/release.yml:47]: What is the purpose of this action?

**Review by Xavrax — APPROVED** [review]
LGTM but I have question

File: .github/workflows/release.yml (hunk 1 above)
Comment (parfeon) [reply .github/workflows/release.yml]: Is it ok now?

Comment (Xavrax) [reply .github/workflows/release.yml]: I didn't requested for change. I was curious what exactly was this action doing. ;d
Anyway made change is also good. ;d

**Review by Xavrax — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (parfeon) [reply .github/workflows/release.yml]: It is an action, which will push to crates and run all our logic with notifications and other stuff :)

**Pull request merged**